osquery/tables/utility/file.cpp

```diff
@@ -33,21 +53,36 @@ const std::map<fs::file_type, std::string> kTypeNames{
     {fs::status_error, "error"},
 };
 
+#endif
+
 void genFileInfo(const fs::path& path,
                  const fs::path& parent,
                  const std::string& pattern,
                  QueryData& results) {
-  // Must provide the path, filename, directory separate from boost path->string
-  // helpers to match any explicit (query-parsed) predicate constraints.
+// Must provide the path, filename, directory separate from boost path->string
+// helpers to match any explicit (query-parsed) predicate constraints.
+
+#if defined(WIN32)
```

Comment:
It looks like there's some sizeable logic here that only applies to Windows. We're ok with doing preprocessor gating to platform, however as we already have our `fileops` abstraction, let's instead move this function to `platformGenFileInfo` inside of the `fileops.cpp` abstraction for both posix and Windows, and then we can avoid needing both logic in the same implementation file.

Reply:
Another possible way to do this might be breaking up this function into smaller bits, so that the osquery `Row` logic stays here, but we generate a `platformStat` function that gets us the data you're seeking.

Reply:
still working on this part.

Reply:
ok what I did is created a `platformStat` function for the Windows fileops that somewhat mirrors the stat function used in POSIX. I kept the `Row` logic in the same file because that seemed cleaner to me. Your thoughts?